src/credentials/tests/FuzzChipCertPW.cpp

```diff
@@ -0,0 +1,95 @@
+#include <cstddef>
+#include <cstdint>
+
+#include <pw_fuzzer/fuzztest.h>
+#include <pw_unit_test/framework.h>
+
+#include "credentials/CHIPCert.h"
+
+namespace {
+
+using namespace chip;
+using namespace chip::Credentials;
+
+using namespace fuzztest;
+
+void ChipCertFuzzer(const std::vector<std::uint8_t> & bytes)
+{
+    ByteSpan span(bytes.data(), bytes.size());
+
+    {
+        NodeId nodeId;
+        FabricId fabricId;
+        (void) ExtractFabricIdFromCert(span, &fabricId);
+        (void) ExtractNodeIdFabricIdFromOpCert(span, &nodeId, &fabricId);
+    }
+
+    {
+        CATValues cats;
+        (void) ExtractCATsFromOpCert(span, cats);
+    }
+
+    {
+        Credentials::P256PublicKeySpan key;
+        (void) ExtractPublicKeyFromChipCert(span, key);
+    }
+
+    {
+        chip::System::Clock::Seconds32 rcacNotBefore;
+        (void) ExtractNotBeforeFromChipCert(span, rcacNotBefore);
+    }
+
+    {
+        Credentials::CertificateKeyId skid;
+        (void) ExtractSKIDFromChipCert(span, skid);
+    }
+
+    {
+        ChipDN subjectDN;
+        (void) ExtractSubjectDNFromChipCert(span, subjectDN);
+    }
+
+    {
+        uint8_t outCertBuf[kMaxDERCertLength];
+        MutableByteSpan outCert(outCertBuf);
+        (void) ConvertChipCertToX509Cert(span, outCert);
+    }
+
+    {
+        // TODO: #34352 To Move this to a Fixture once Errors related to FuzzTest Fixtures are resolved
```

Comment:
I am looking at https://github.com/project-chip/connectedhomeip/issues/34352 and don't see a description of `FuzzTest Fixture errors`. Is there a different issue or could we create a separate issue explaining what Fuzztest Fixture errors mean and what the resolution would be?

Reply:
New Issue created: #35369